gui-wm/wayfire/wayfire-0.4.0.ebuild

```diff
@@ -0,0 +1,97 @@
+# Copyright 2019-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit meson
+
+DESCRIPTION="compiz like 3D wayland compositor"
+HOMEPAGE="https://github.com/WayfireWM/wayfire"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/WayfireWM/${PN}.git"
+	KEYWORDS=""
+else
+	SRC_URI="https://github.com/WayfireWM/${PN}/releases/download/${PV}/${P}.tar.xz"
+	KEYWORDS="~amd64 ~arm64 ~x86"
+fi
+
+LICENSE="MIT"
+SLOT="0"
+IUSE="+gles2 +wf-config +wlroots +elogind systemd debug"
+
+DEPEND="
+	dev-libs/libevdev
+	dev-libs/libinput
+	gui-libs/gtk-layer-shell
+	media-libs/glm
+	media-libs/mesa:=[gles2,wayland,X]
+	media-libs/libjpeg-turbo
+	media-libs/libpng
+	media-libs/freetype:=[X]
+	x11-libs/libdrm
+	x11-libs/gtk+:3=[wayland,X]
+	x11-libs/cairo:=[X,svg]
+	x11-libs/libxkbcommon:=[X]
+	x11-libs/pixman
+	gles2? ( media-libs/libglvnd[X] )
+	wf-config? ( ~gui-libs/wf-config-${PV}[debug=] )
+	!wf-config? ( !gui-libs/wf-config )
+	wlroots? ( >=gui-libs/wlroots-0.10.0[elogind=,systemd=,X] )
```

Comment:
is wlroots not a hard dependency for Wayfire? I am confused by this logic here along with the wf-config logic.

Reply:
It is a hard dependency but wayfire also has another method to install wlroots by pulling it in as a submodule and then building that and installing it system wide.  
The advantage of that is that wayfire will always pull the most compatible version of wlroots (down the commit) as is mandated by upstream.  
The disadvantage is that this conflicts with the system wide wlroots install.  
Upstream recommended to use the submodule version for best compatibility.  
But I've made it so the nearest compatible version is the one required in the ebuild.

This is a weird situation but I recommend having this option as it makes debugging a lot easier, when you are almost sure that the error is not due to an incompatibility between wayfire and wlroots.

Reply:
same for wf-config as well.
This is very useful for the live ebuild testing versions.